src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```
@@ -8453,6 +8507,128 @@ String replaceParameterMarkers(String sqlSrc, int[] paramPositions, Parameter[]
+                return "CAST(" + plainStr + " AS DECIMAL(" + sqlPrecision + "," + sqlScale + "))";
+            }
+            return plainStr;
+        }
+
+        // Integer types can use toString() safely
+        if (value instanceof Byte || value instanceof Short || value instanceof Integer || value instanceof Long) {
+            return value.toString();
+        }
+
+        // Float and Double need special handling to avoid precision loss and scientific
+        // notation
+        if (value instanceof Float) {
+            // Convert Float to BigDecimal for precise representation
+            return new java.math.BigDecimal(value.toString()).toPlainString();
+        }
+
+        if (value instanceof Double) {
+            // Convert Double to BigDecimal for precise representation
+            return new java.math.BigDecimal(value.toString()).toPlainString();
+        }
+
+        if (value instanceof java.sql.Date) {
+            return "CAST('" + DATE_FMT.format((java.util.Date) value) + "' AS DATE)";
+        }
+
+        if (value instanceof java.sql.Time) {
+            return "CAST('" + TIME_FMT.format((java.util.Date) value) + "' AS TIME)";
+        }
+
+        if (value instanceof java.sql.Timestamp) {
+            return "CAST('" + TS_FMT.format((java.util.Date) value) + "' AS DATETIME2)";
+        }
+
+        if (value instanceof java.util.Date) {
+            // generic java.util.Date -> timestamp
+            return "CAST('" + TS_FMT.format((java.util.Date) value) + "' AS DATETIME2)";
+        }
+
+        if (value instanceof byte[]) {
+            return bytesToHexLiteral((byte[]) value);
+        }
+
+        if (value instanceof Blob) {
+            try {
+                Blob b = (Blob) value;
+                int len = (int) b.length();
+                byte[] bytes = b.getBytes(1, len);
+                return bytesToHexLiteral(bytes);
+            } catch (Exception e) {
+                return "NULL";
```

We should not catch these exceptions and let them be thrown back to application, returning `NULL` string silently will cause wrong results